### PR TITLE
Update welcome emails to use new HTML library

### PR DESCRIPTION
**Pull request by mbrandonw**

### CODE REVIEW

File: Sources/PointFree/Admin/EmailPreview.swift [hunk 1]
```diff
@@ -42,14 +42,13 @@ private func emailPreviewView(selectedTemplate: EmailTemplate?) -> Node {
         .fragment(options(selectedTemplate: selectedTemplate))
       )
     ),
-    .div(
+    .iframe(
```
Comment (mbrandonw) [inline Sources/PointFree/Admin/EmailPreview.swift:45]: Better to render email previews in its own iframe so it's fully quarantined from the styles on this page.

File: Sources/PointFree/Emails/SendEmail.swift [hunk 2]
```diff
@@ -91,6 +92,67 @@ public func prepareEmail(
   )
 }
 
+public func prepareEmailV2(
```
Comment (mbrandonw) [inline Sources/PointFree/Emails/SendEmail.swift:95]: Once we convert the rest of the emails to this new system we can get rid of `prepareEmail` in favor of this.

File: Sources/PointFree/Emails/WelcomeEmails.swift [hunk 3]
```diff
@@ -0,0 +1,271 @@
+import Dependencies
+import Models
+import IssueReporting
+import PointFreeRouter
+import StyleguideV2
+
+struct WelcomeEmail<Content: HTML>: EmailDocument {
```
Comment (mbrandonw) [inline Sources/PointFree/Emails/WelcomeEmails.swift:7]: This is a general "layout" for welcome emails, and could probably be generalized for all emails.

File: Sources/PointFree/Emails/WelcomeEmails.swift [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+  ) {
+    self.content = content()
+    self.preheader = preheader
+    self.user = user
+  }
+
+  var body: some HTML {
+    span {
+      HTMLText(preheader)
+    }
```
Comment (mbrandonw) [inline Sources/PointFree/Emails/WelcomeEmails.swift:27]: "Preheader" is a bit of hidden text you put at the top of emails that most email clients will render as the one-line preview in the inbox.

File: Sources/PointFree/RunnerTasks/SendWelcomeEmails.swift [hunk 5]
```diff
@@ -0,0 +1,143 @@
+import Dependencies
```
Comment (mbrandonw) [inline Sources/PointFree/RunnerTasks/SendWelcomeEmails.swift:1]: This file looks new in the diff but it's just me renaming an existing file.

File: Sources/StyleguideV2/HTMLHelpers/Link.swift [hunk 6]
```diff
@@ -19,6 +19,7 @@ public struct Link<Label: HTML>: HTML {
   public var body: some HTML {
     a { label }
       .attribute("href", href)
+      .color(linkStyle.color)
```
Comment (mbrandonw) [inline Sources/StyleguideV2/HTMLHelpers/Link.swift:22]: This is necessary to get Gmail to recognize link color, but it also causes a lot of churn in our snapshot tests. We could think about ways to limit this only to email html, but not sure how we want to do that.

File: Sources/PointFree/Admin/EmailPreview.swift [hunk 7]
```diff
@@ -74,11 +73,32 @@ private func email(selectedTemplate: EmailTemplate) -> Node {
       newPricing: Pricing(billing: .yearly, quantity: 7)
     )
   case .welcomeEmail1:
-    return welcomeEmail1Content(user: blob)
+    return Node {
+      WelcomeEmail(
+        preheader: "This is the preheader",
```
Comment (stephencelis) [inline Sources/PointFree/Admin/EmailPreview.swift:78]: - [ ] TODO

File: Sources/PointFree/Emails/SendEmail.swift [hunk 8]
```diff
@@ -91,6 +92,67 @@ public func prepareEmail(
...
+  unsubscribeData
+    .map { userId, newsletter in
+      guard
+        let unsubEmail = mailgun.unsubscribeEmail(
+          fromUserId: userId, andNewsletter: newsletter),
+        let unsubUrl = (try? ExpressUnsubscribe().print((userId, newsletter)))
+          .flatMap({ Encrypted(String($0), with: envVars.appSecret) })
+          .map({ siteRouter.url(for: .expressUnsubscribe(payload: $0)) })
+      else {
+        logger.log(.error, "Failed to generate unsubscribe link for user \(userId)")
```
Comment (stephencelis) [inline Sources/PointFree/Emails/SendEmail.swift:124]: Should this use `reportIssue`?

File: Sources/StyleguideV2/HTML/HTMLPrinter.swift [hunk 9]
```diff
@@ -31,19 +31,25 @@ public struct HTMLPrinter {
...
+          sheet.append("\(className){\(style)}")
+        }
         sheet.append(configuration.newline)
       }
     }
     return sheet
   }
 
   public struct Configuration {
+    let forceImportant: Bool
```
Comment (stephencelis) [inline Sources/StyleguideV2/HTML/HTMLPrinter.swift:46]: We could consider calling this `isHTMLEmail` or `isEmail` instead to document why, but we also know we wanna reconsider the HTML library design with more use and after the dust settles.

**Review by stephencelis — APPROVED** [review]

**Pull request merged**